internal/util/filesystem.go

```diff
@@ -18,27 +18,35 @@ type FileSystemUtil struct {
 }
 
 func (u *FileSystemUtil) writeToYAMLFile(path string, _struct interface{}) error {
-
+	var deferredErr error = nil
 	_yaml, err := yaml.Marshal(_struct)
 	if err != nil {
 		return err
 	}
 
 	if !u.IsExistingFile(path) {
-		os.MkdirAll(u.GetConfigDir(), os.FileMode(0700))
+		err := os.MkdirAll(u.GetConfigDir(), os.FileMode(0700))
+		if err != nil {
+			return err
+		}
 		var file *os.File
 		file, err = os.Create(path)
 		if err != nil {
 			return err
 		}
-		defer file.Close()
+		defer func() {
+			err := file.Close()
```

Comment:
Does this `deferredErr` thing work? 

Can we just ignore these uncaught errors (either with a //nolint comment, or with ` _ = file.Close()` call?)

Reply:
Nope, looks like I missed that